scripts/common-functions.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+# Check platform
+platform=$(uname)
+
+# Check and install missing packages
+if [[ $platform == "Darwin" ]]; then
```

Comment:
seems we want to support both `Darwin` and `Linux` platform, do you think we need to test those common functions e.g. slackNotification() against Darwin platform ? to check if those cli options (e.g. jq, curl, etc) are compatible or not

Reply:
There are multiple files in this repo that have this check/function and the goal of this script is to reduce duplication and improve consistency across the repo.
This function in aprticular is included for future PRs where those scripts will be changed to make use of this script.

Reply:
Pretty sure the only reason we have this Darwin check is enable local running on Mac OS as it uses gdate instead of date. I personally wouldn't worry about the Darwin piece too much.